conductor-client/src/main/kotlin/com/openlattice/authorization/mapstores/ExternalPermissionRolesMapstore.kt

```diff
@@ -0,0 +1,57 @@
+package com.openlattice.authorization.mapstores
+
+import com.openlattice.authorization.AccessTarget
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.mapstores.TestDataFactory
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.postgres.PostgresTable
+import com.openlattice.postgres.ResultSetAdapters
+import com.openlattice.postgres.mapstores.AbstractBasePostgresMapstore
+import com.zaxxer.hikari.HikariDataSource
+import org.springframework.stereotype.Component
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.util.*
+
+/**
+ * @author Drew Bailey (drew@openlattice.com)
+ */
+@Component
+class ExternalPermissionRolesMapstore(
+        val hds: HikariDataSource
+) : AbstractBasePostgresMapstore<AccessTarget, UUID>(
+        HazelcastMap.EXTERNAL_PERMISSION_ROLES, PostgresTable.EXTERNAL_PERMISSION_ROLES, hds
+) {
+    override fun generateTestKey(): AccessTarget {
+        return AccessTarget(
+                TestDataFactory.role().aclKey,
+                TestDataFactory.permissions().first()
+        )
+    }
+
+    override fun generateTestValue(): UUID {
+        return UUID.randomUUID()
+    }
+
+    override fun bind(ps: PreparedStatement, key: AccessTarget, value: UUID) {
+        var index = bind(ps, key, 1)
+        ps.setObject(index++, value)
+
+        ps.setObject(index++, value)
```

Comment:
Yes, this is the pattern in all Mapstores, the second one is needed because all mapstore writes are upserts so they take the various parameters twice, once for the insert, once for the on conflict